docs/source/developers/bug_reports.rst

```diff
@@ -186,10 +176,109 @@ Good reproducible examples or minimal bug reports can be found in next tabs:
 * `R: Tidyverse: Make a reprex <https://www.tidyverse.org/help/#reprex>`_
 * `R: Tidyverse's Reprex do's and don'ts <https://reprex.tidyverse.org/articles/reprex-dos-and-donts.html>`_
 * `Mozilla's bug-reporting guidelines <https://developer.mozilla.org/en-US/docs/Mozilla/QA/Bug_writing_guidelines>`_
+
+.. _identify-component:
+
+Identify Arrow component
+++++++++++++++++++++++++
+
+Arrow is an expansive project supporting many languages and organized into a
+number of components. Identifying the affected component(s) helps new issues
+get attention from appropriate contributors.
+
+* Use the **Component** field to indicate the area of the project that your
+  issue pertains to (for example "Python" or "C++").
+* Also prefix the issue title with the component name in brackets, for example
+  ``[Python] issue name`` ; this helps when navigating lists of open issues,
+  and it also makes our changelogs more readable. Most prefixes are exactly the
+  same as the **Component** name, with the following exceptions:
+
+  * **Component:** Continuous Integration — **Summary prefix:** [CI]
+  * **Component:** Developer Tools — **Summary prefix:** [Dev]
+  * **Component:** Documentation — **Summary prefix:** [Docs]
+
+.. _affected-version:
+
+Identify affected version
++++++++++++++++++++++++++
+
+If you're reporting something that used to work in a previous version
+but doesn't work in the current release, you can add the **Affects version**
+field to identify the earliest known version where the bug is observed.
+For feature requests and other proposals, leave **Affects version** empty as
+it is not applicable.
+
+.. _issue-lifecycle:
+
+Issue lifecycle
+===============
+
+Both bug reports and feature requests follow a defined lifecycle. The issue
+**Status** field is used to document the current state of the issue, while the
+**Resolution** field indicates the outcome of issues that have reached
+terminal status. 
+
+
+.. _issue-status:
+
+Issue Status
+++++++++++++
+
+The Arrow project uses the following statuses in Jira to indicate what has -
+and will be - done on an issue:
+
+* **Open** - This is the initial issue state, prior to a contributor assigning
+  the issue and starting progress. Issues in this state should be unassigned.
+* **In progress** - At the time a contributor self-assigns an issue, the status
+  should be set to In progress by clicking the **Start progress** button. All
+  issues in this status should have an assignee - unassigned issues will be
+  set back to a status of Open. Issues remain "in progress" until resolved or
+  closed, including during review of pull requests.
+* **Resolved** - This is a terminal status indicating action has been taken
+  on the issue, which is now considered completed. Issues in a resolved status
+  may have the following resolution codes set:
+
+  * Fixed
+  * Implemented
+  * Done
+
+* **Closed** - Another terminal status, Closed indicates the issue is complete,
```

Comment:
**_NOTE:_** Resolution codes have not been consistently applied historically. See [this analysis ](https://docs.google.com/spreadsheets/d/1FsGIHTRN20Mq3AhS8Wuiwmf9-bBdNeCTVjgGf7vfKHA/edit?usp=sharing)for more details, and to comment on proposal to standardized and clean existing resolution code data. The resolution codes listed here eliminate several codes that have been previously used, and aims to standardize usage for clarity.

Reply:
Right, the problem here is there are too many possible choices with no one being obviously better than the others in some situations :-)